pkg/driver/node_server.go

```diff
@@ -1621,6 +1623,28 @@ func (driver *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoReque
 		nodeMaxVolumesLimit = defaultMaxVolPerNode
 	}
 
+	// Enable watcher only once. GetNodeInfo rpc may be called multiple times by external
+	// provisioner.
+	if !isWatcherEnable {
+		// Create a anonymous wrapper function over nodeGetInfo. This is os event driven
+		// fn execution.
+		getNodeInfoFunc := func() {
+			nodeID, err := driver.nodeGetInfo()
+			if err != nil {
+				log.Errorf("Failed to update %s nodeInfo. Error: %s", nodeID, err.Error())
+			}
+			return
+		}
+		// Register anonymous wrapper function(watcher).
+		csiWatch, _ := driver.InitializeWatcher(getNodeInfoFunc)
+		// Add list of files /and directories to watch. The list contains
+		// iSCSI , FC and CHAP Info and Networking config directories
+		list := []string{"/etc/sysconfig/network-scripts/", "/etc/sysconfig/network/", "/etc/iscsi/", "/sys/class/fc_host/"}
+		csiWatch.AddWatchList(list)
+		// Start event the watcher in a separate thread.
+		go csiWatch.StartWatcher()
```

Comment:
Per @shivamerla , how are we handling spurious updates?  We don't want the getNodeInfoFunc to be called on every update to every file in these directories.

Reply:
I am capturing all the events trigger on WatchList, However, we could go even more granular in our implementation like only check for CREATE event in network directory. Please elaborate on spurious updates so that we can safeguard against it.

Reply:
This is along the lines of rate limiting.  We don't want to call nodeGetInfo on every update.  Only on all updates within a time interval.

Reply:
Got it. thanks.